src/source-pcap-file-helper.c

```diff
@@ -105,12 +132,28 @@ void PcapFileCallbackLoop(char *user, struct pcap_pkthdr *h, u_char *pkt)
 
     PACKET_PROFILING_TMM_END(p, TMM_RECEIVEPCAPFILE);
 
+    /* What TmThreadsSlotVarRun did for the decode module. */
+    if (DecodePcapFile(ptv->shared->tv, p, ptv->shared->dtv) != TM_ECODE_OK) {
+        TmThreadsSlotProcessPktFail(ptv->shared->tv, ptv->shared->slot, NULL);
+        goto fail;
+    }
+
+    /* Handle pseudo packets created in decode. */
+    if (TmThreadsProcessDecodePseudoPackets(
+                ptv->shared->tv, &ptv->shared->tv->decode_pq, ptv->shared->slot) != TM_ECODE_OK) {
+        goto fail;
+    }
```

Comment:
I don't like how every receive module will need this, but it can't be moved into TmThreadsSlotProcessPkt without some refactoring there, but still looking into this. Open to thoughts.